docs/pairing-architecture.md

```diff
@@ -0,0 +1,283 @@
+It's PII, but the channel has established a base level of trust between devices, so it should be safe to share this information directly.
+
+* Upon receiving the metadata message from the other device, both Desktop and Mobile show an explicit confirmation screen. 
+* There's no set confirmation order: the Supp and Auth will show the confirmations screens at the same time, and can be accepted in any order.
+
+For additional security, the Channel Server annotates each message with information about the originating device, such as geoip lookup data. This may help the user detect if they're connecting to a different device than expected.
+
+For additional security, each device can display the calculated confirmation code to the user. This may help the user detect if they're connecting to a different device than expected.
+
+#### 4) Desktop device authorizes OAuth request
+
+When the user accepts the confirmation screen on the Desktop device, it will use its local credentials to authorize the OAuth request from the Mobile device. This involves no user interaction, and will include the following steps:
+
+* Check that redirect_uri is the expected value; error out if not.
+* Check that scope contains exactly `"profile"` and `"https://identity.mozilla.com/apps/oldsync"`; error out if not.
+* Check that client_id is the expected value; error out if not.
+* Check that state exists and is well formed; error out if not.
+* Check that code_challenge and code_challenge_method exist and are well formed; error out if not.
+* Build the scoped-key bundle for `"https://identity.mozilla.com/apps/oldsync"`, encrypting it with the provided `keys_jwk` to produce `keys_jwe`.
+* Generate a BrowserID assertion with the fxa-oauth-server as audience.
+* POST all of this to fxa-oauth-server at `/v1/authorization` and receive back an OAuth code and state response parameters.
+
+Desktop will then send the code and state parameters back to Mobile over the websocket channel, encrypted as before with the channel key.
+
+#### 5) Mobile receives OAuth response and connects to Sync
+
+When the user accepts the confirmation screen on the Mobile device, it will listen for the message produced by the Desktop device in step (4) above. After receiving the message, it can tear down its websocket channel.
+
+* The Mobile can now complete its OAuth flow just as if it had received code and state via a standard web-content-based OAuth flow:
+* Submit the code to fxa-oauth-server at /v1/token in exchange for an access token, refresh token, and bundle of encrypted key material.
+* Decrypt the bundle of key material with its keys_jwk private key.
+* Fetch user profile data, and check that it matches the data provided by Desktop in step (3).
+* Use the access token and key material to access Firefox Sync.
+
+
+## Exchanged messages specification
+
+### Pairing Channel messages
+These messages are forwarded between the Desktop native code and the web content page opened on the supplicant device.
```

Comment:
It's not clear that this doc has appropriate introduced the notion of "web content page opened on the supplicant device", it might need a paragraph of two in the introduction to say more about the moving parts involved here.